src/AzureOperator.ts

```diff
@@ -68,24 +40,9 @@ export class AzureOperator {
       return;
     }
 
-    let iotProject;
-    telemetryContext.properties.projectHostType =
-        ProjectHostType[this.projectHostType];
-    if (this.projectHostType === ProjectHostType.Container) {
-      iotProject = new ioTContainerizedProjectModule.IoTContainerizedProject(
-          context, channel, telemetryContext);
-    } else if (this.projectHostType === ProjectHostType.Workspace) {
-      iotProject = new ioTWorkspaceProjectModule.IoTWorkspaceProject(
-          context, channel, telemetryContext);
-    }
-    if (iotProject === undefined) {
-      await handleIoTWorkspaceProjectFolder(telemetryContext);
-      return;
-    }
-
-    const result = await iotProject.load();
-    if (!result) {
-      await askAndNewProject(telemetryContext);
+    const iotProject =
+        await createAndLoadIoTProject(context, channel, telemetryContext);

```

Comment:
Does deploy need create project?

Reply:
Will change the function name to `constructAndLoadIoTProject`.